src/components/mixins/SplashMixin.vue

```diff
@@ -0,0 +1,93 @@
+<script lang='ts'>
+import Vue from 'vue';
+import Component from 'vue-class-component';
+
+import ApiResponse from '../../model/api/ApiResponse';
+import GameManager from '../../model/game/GameManager';
+import RequestItem from '../../model/requests/RequestItem';
+import ConnectionProvider from '../../providers/generic/connection/ConnectionProvider';
+import ThunderstorePackages from '../../r2mm/data/ThunderstorePackages';
+import ApiCacheUtils from '../../utils/ApiCacheUtils';
+
+@Component
+export default class SplashMixin extends Vue {
```

Comment:
What do we gain by making this a mixin?
If the point was to decouple logic from the template, I'd argue that it hasn't really separated any logic from the page, just moved it.

Reply:
This change was mainly for reducing duplication between r2mm and tsmm, and hopefully making maintaining both easier in the future. It does clean up the [tsmm side](https://github.com/thunderstore-io/thunderstore-mod-manager/pull/109/commits/902574c972b99993de38456f2d4a5b78fce9adc1) quite nicely IMO.